data/campaigns/Under_the_Burning_Suns/utils/abilities.cfg

```diff
@@ -356,6 +293,36 @@ Any units adjacent to this unit will fight as if it were dusk when it is night,
                 special_id=sun_incarnate
             [/has_attack]
         [/filter]
+        [event]
+            name = attack_end
+            first_time_only = no
+            id = song_verse_illuminates
+            [filter_attack]
+                name = sun incarnate
+            [/filter_attack]
+            [object]
+                id = song_$unit.underlying_id|_$turn_number
+                [effect]
+                    apply_to=attack
+                    name=sun incarnate
+                    [set_specials]
+                        mode=append
+                        [dummy]
+                            id=sun_incarnate
+                        [/dummy]
+                    [/set_specials]
+                [/effect]
+                duration = scenario
+            [/object]
+            [event]
+                name = "side $side_number turn $($turn_number + 3)"
```

Comment:
I don't want to set up tests right now, so just asking if this has been tested - if a unit with this attack is placed during an event, do these variables get interpolated at the correct time?

Reply:
Did some testing, and it works.